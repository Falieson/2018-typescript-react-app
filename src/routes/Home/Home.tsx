import Gx from 'gx'
import * as React from 'react'
import Helmet from 'react-helmet'

import Hello from '../../components/hello'

function HomeEntrancePage() {
  return (
    <div>
      <Helmet>
        <title>Homepage</title>
      </Helmet>

      <Gx col={12}>
        <div className="rtv-content--entrance">
          <h2>
            <Hello />
          </h2>
          <p>
            Set the info you care most about...
          </p>
        </div>
      </Gx>
    </div>
  )
}

export default HomeEntrancePage
