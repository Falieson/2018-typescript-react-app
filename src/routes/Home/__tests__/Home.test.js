/* eslint-disable import/no-extraneous-dependencies */

import React from 'react'
import { shallow } from 'enzyme'

import Home from '../Home'

describe('<Home />', () => {
  test('renders paragraph text', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find('p').length).toBe(1)
  })
  test('renders h2 element', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper.find('h2').length).toBe(1)
  })
  test('renders', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()
  })
})
