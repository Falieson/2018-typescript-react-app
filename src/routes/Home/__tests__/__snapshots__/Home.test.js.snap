// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Dashboard
    </title>
  </HelmetWrapper>
  <Gx
    col={12}
  >
    <div
      className="rtv-content--entrance"
    >
      <h2>
        Welcome to your dashboard...
      </h2>
      <p>
        Track the polls you care about...
      </p>
    </div>
  </Gx>
</div>
`;
